app/main.py

```diff
@@ -1,3 +1,61 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    @staticmethod
+    def __typecheck__(other: any) -> None:
+        return isinstance(other, Distance)
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: any) -> float:
+        if Distance.__typecheck__(other):
+            return Distance(self.km + other.km)
+        other = Distance(other)
+        return Distance(self.km + other.km)
```

Comment:
you can just `return Distance(self.km + other)` instead of `$21-22` lines. WDYT?